index.mjs

```diff
@@ -1,10 +1,11 @@
+const midCache = new Map(); // midpoint vertices cache to avoid duplicating shared vertices
 
 export default function icomesh(order = 4) {
     if (order > 10) throw new Error(`Max order is 10, but given ${order}.`);
 
     // set up an icosahedron (12 vertices / 20 triangles)
     const f = (1 + Math.sqrt(5)) / 2;
-    const T = Math.pow(4, order);
+    const T = 1 << (order << 1); // Math.pow(4, order);
 
     const vertices = new Float32Array((10 * T + 2) * 3);
     vertices.set([
```

Comment:
Should we also do `Float32Array.of` here for consistency?

Reply:
Hmm, what you mean? Something like this?
```js
const vertices = new Float32Array((10 * T + 2) * 3);
vertices.set(Float32Array.of(
  -1, f, 0, 1, f, 0, -1, -f, 0, 1, -f, 0,
  0, -1, f, 0, 1, f, 0, -1, -f, 0, 1, -f,
  f, 0, -1, f, 0, 1, -f, 0, -1, -f, 0, 1
));
```

Reply:
Yes, exactly. Although I'm not sure whether this is more optimal — v8 generally doesn't like polymorphic methods, especially with a ton of arguments, right?